lib/utilities/project-name.js

```diff
@@ -1,5 +1,21 @@
 'use strict';
 
+const INVALID_PROJECT_NAMES = ['test', 'ember', 'ember-cli', 'vendor', 'public', 'app', 'addon', 'application'];
+
+function sanitizeProjectName(name) {
+  let isScoped = name[0] === '@' && name.includes('/');
+
+  let sanitizedName = name;
+
+  if (isScoped) {
+    sanitizedName = sanitizedName
+      .slice(1) // drop the `@`
+      .replace('/', '-'); // replace the first / char with -
```

Comment:
I would think people would want just the right side of the / for the folder name. My idea is you are already in your scope dir that has all your addons (ie "/projects/my-scope"), and you make your addons via `ember new @my-scope/my-addon-2`, you expect a project name of `@my-scope/my-addon-2` with the project structure "/projects/my-scope/my-addon-2", not "projects/my-scope/my-scope-my-addon-2". I think this makes sense in a monorepo too, where your monorepo is the scope and your packages dirs don't have the scope in their basename. I guess it all comes down to how people organise their scopes, but this is not how I'd expect it to work.

Reply:
Hmm, for all scenarios where I've created a scoped package I've used this format (https://github.com/emberjs/ember-optional-features, https://github.com/emberjs/ember-edition-utils, https://github.com/emberjs/ember-test-helpers). I agree that within a mono-repo this would be a bit different (you'd likely make `packages/@scope-name/project-name`).

In this case, we are only providing the default (and I just picked the default that I've seen and personally used before). The escape hatch here is that if the user specifies a specific `--directory` then we will use that value instead of inferring one.

What is the path forward here?

Reply:
In addition to the above, the report from [here](https://github.com/ember-cli/ember-cli-update/issues/464) also has this layout `@html-next/flexi-config` => "/Users/rwwagner/html-next/flexi-config/"

Reply:
@kellyselden - So if we detect that `path.basename(process.cwd()) === scopeName` then we do _not_ prefix, otherwise we do?

Reply:
I vote for always strip off scope, and escape to --directory if you want scope in the path. If I'm scaffolding a monorepo (ie /my-scope/packages/), it would fail that cwd check.

Reply:
Hmm, don't most mono-repos that are actually publishing scoped packages include the scoped folder name!? Some examples:

* https://github.com/glimmerjs/glimmer-vm/tree/master/packages
* https://github.com/glimmerjs/glimmer.js/tree/master/packages
* https://github.com/emberjs/ember.js/tree/master/packages
* https://github.com/vuejs/vue-cli/tree/dev/packages

Also, here are some popular monorepo packages that actually use `scope-name-package-name` in repo (what this PR would do now):

* https://github.com/babel/babel/tree/master/packages
* https://github.com/zeit/next.js/tree/canary/packages

I think the fact of the matter is that there just _isn't_ a standard way of doing this.

---

@kellyselden - FWIW, I don't particularly think that either of us are going to change our opinions on a default here, but we need to come to some sort of compromise because the current situation is massively more horrible than having to `mv scope-name-package-name wherever-i-actually-want-it`.

Reply:
Let's go with your compromise of checking cwd, otherwise dasherize. Since you're doing the work, you get more sway :)

Reply:
fwiw we don't include the scoped folder name in EmberData. Maybe we should? Seems like an unnecessary extra folder if you're already in a monorepo for the scope.

Reply:
(Also I didn't realize it was common to have the extra folder, I am now less surprised I find navigating the code for so many other workspaces including Ember's so arduous, I'd always thought Ember's was that way to handle migrating from legacy)

Reply:
When not using a monorepo for scoped I'd say I usually don't have the `@` in the folder path. For instance with `html-next` (the github org and npm scope mentioned above), the repo will be `html-next/foo` and the package name will be `@html-next/foo`. Unlike the comment above on having the `@scope/` typically be part of the path in `packages` for a monorepo, I feel for the non-monorepo case this `org/name` structure is far more common.

Reply:
@runspired my suggestion in https://github.com/ember-cli/ember-cli/pull/8964#discussion_r359464747 would have required `html-next` **not** `@html-next` (the `@` is not part of the "scope name" IMHO, it's mostly only a sigil used to denote that it is a scope). If we tweak that _slightly_ to look for _any_ parent directory including the scope name, I think that would also handle the html-next and ember-data case (with the exception of the outlier `packages/-ember-data` folder) as well as @kellyselden's original point in https://github.com/ember-cli/ember-cli/pull/8964#discussion_r359430711.

Reply:
OK, updated to implement my suggestion just above. If any directory including or above `process.cwd()` are _exactly_ the scope name, the new folder created will be the package name without scope other wise it will be `scope-name-package-name`.

Reply:
Sorry to suggest another change, but in one of your examples above (https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue), the scope folder has @. Do you think we could/should check for without or _with_ the @?